### PR TITLE
added python-based dds converter

### PR DESCRIPTION
**Pull request by zemmyang**
tested it a bit on windows

### CODE REVIEW

File: tools/dds.py [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+        '<4I4x',
+        dwCaps1(),  # dwCaps1
+        0,  # dwCaps2 - no cubemaps in DXHR... technically.
+        0,  # dwCaps3 = 0
+        0  # dwCaps4 = 0
+    )
+
+    assert len(out_blob) == 124 + 4
+
+    out_blob = data[28:]
```
Comment (rrika) [inline tools/dds.py]: how come out_blob is discarded here? should this be `+=` perhaps

File: tools/cdcmesh.py [hunk 2]
```diff
@@ -526,47 +526,50 @@ def read_skeleton(skeletonblob, bonecount):
...
+	# else:
+	# 	firstmip = data[0x1C:0x1C+width*height*4]
+	# 	pixels = firstmip
 
 	im = bpy.data.images.new(basename, width, height)
-	pi = numpy.frombuffer(pixels, dtype="B")
+	pi = numpy.frombuffer(out_blob, dtype="B")
 	po = numpy.empty(shape=pi.shape, dtype=float)
 	numpy.multiply(pi, 1/255.0, po)
 	im.pixels = po.data
```
Comment (rrika) [inline tools/cdcmesh.py]: I would've assumed that the new blob is passed into blender differently (not by im.pixels=), and that the numpy dependency and the division by 255 become superfluous.

**Review by rrika — CHANGES_REQUESTED** [review]

File: tools/dds.py (hunk 1 above)
Comment (zemmyang) [reply tools/dds.py]: fixed that in the 2nd commit

File: tools/cdcmesh.py (hunk 2 above)
Comment (zemmyang) [reply tools/cdcmesh.py]: you're right

i did have a fix for that in my second commit, but it uses PIL (and im only learning now that it's not shipped with blender by default). 

currently looking into it

**Comment by zemmyang** [comment]
ok, 3rd push just saves the DDS files to the tempdir, then loads the image back, then packed into the blend file. images are only created once based on the section number. tested it on a couple of large IMFs using a fresh blender 3.0 install on win10, works on my end

**Pull request merged**